dev/io.openliberty.netty.internal.tls.impl/src/io/openliberty/netty/internal/tls/impl/NettyTlsProviderImpl.java

```diff
@@ -0,0 +1,329 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.netty.internal.tls.impl;
+
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+import java.util.*;
+import java.util.Map.Entry;
+
+import javax.net.ssl.*;
+
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ssl.*;
+
+import io.netty.handler.ssl.*;
+import io.openliberty.netty.internal.tls.NettyTlsProvider;
+import io.netty.handler.ssl.SslContext;
+
+/**
+ * Creates {@link io.netty.handler.ssl.SslContext} objects via
+ * {@link io.netty.handler.ssl.JdkSslContext.JdkSslContext} using active Liberty
+ * SSL configurations
+ * 
+ * TODO: this logic should be made generic and put in a separate bundle so that
+ * the SIP feature does not have a SSL runtime dependency
+ * 
+ * Adapted from
+ * {@link com.ibm.ws.channel.ssl.internal.SSLChannel.getSSLContextForLink(VirtualConnection,
+ * String, String, String, Boolean, SSLConnectionLink)}
+ *
+ */
+@Component(configurationPid = "io.openliberty.netty.internal.tls", immediate = true, service = NettyTlsProvider.class, property = { "service.vendor=IBM" })
+public class NettyTlsProviderImpl implements NettyTlsProvider {
+
+    private static final TraceComponent tc = Tr.register(NettyTlsProvider.class);
```

Comment:
Should this be NettyTlsProvider.class or NettyTlsProviderImpl.class?

Reply:
updated